pkg/config/system_probe.go

```diff
@@ -165,6 +159,16 @@ func InitSystemProbeConfig(cfg Config) {
 
 	// service monitoring
 	cfg.BindEnvAndSetDefault(join(smNS, "enabled"), false, "DD_SYSTEM_PROBE_SERVICE_MONITORING_ENABLED")
+
+	// runtime compilation settings
+	cfg.BindEnvAndSetDefault(join(rcNS, "runtime_compiler_output_dir"), defaultRuntimeCompilerOutputDir, "DD_RUNTIME_COMPILER_OUTPUT_DIR")
+	cfg.BindEnvAndSetDefault(join(rcNS, "enable_kernel_header_download"), false, "DD_ENABLE_KERNEL_HEADER_DOWNLOAD")
+	cfg.BindEnvAndSetDefault(join(rcNS, "kernel_header_dirs"), []string{}, "DD_KERNEL_HEADER_DIRS")
+	cfg.BindEnvAndSetDefault(join(rcNS, "kernel_header_download_dir"), defaultKernelHeadersDownloadDir, "DD_KERNEL_HEADER_DOWNLOAD_DIR")
+	cfg.BindEnvAndSetDefault(join(rcNS, "apt_config_dir"), suffixHostEtc(defaultAptConfigDirSuffix), "DD_APT_CONFIG_DIR")
+	cfg.BindEnvAndSetDefault(join(rcNS, "yum_repos_dir"), suffixHostEtc(defaultYumReposDirSuffix), "DD_YUM_REPOS_DIR")
+	cfg.BindEnvAndSetDefault(join(rcNS, "zypper_repos_dir"), suffixHostEtc(defaultZypperReposDirSuffix), "DD_ZYPPER_REPOS_DIR")
+	cfg.BindEnvAndSetDefault(join(rcNS, "skip_runtime_compilation"), false, "DD_SKIP_RUNTIME_COMPILATION")
```

Comment:
How do `enable_runtime_compliation` and `skip_runtime_compilation` relate? Why do we need this one?

Reply:
`enable_runtime_compliation` is a module-specific setting which only applies to assets within the module - i.e. runtime security has an `enable_runtime_compliation` setting which enables compilation of the runtime security probe & the constant fetcher, and NPM has an `enable_runtime_compliation` setting which enables compilation of the network tracer, http tracer and ebpf conntracker.
`skip_runtime_compilation` skips the entire runtime compilation module, so it doesn't matter what other settings are enabled, runtime compilation will not occur when the system-probe is run.
The purpose of this combination of settings is that you can configure specifically which assets you want to enable runtime compilation for,  set `skip_runtime_compilation` to `true` and run the system-probe's `compile` command (i.e. in an init container) and compilation will occur as expected. Then, when you run the system-probe afterwards, runtime compilation will not occur, but the assets which you have enabled runtime compilation for will try and load the runtime compiled `.o` files

Reply:
That makes sense, but the naming is a little confusing. But I don't have a better suggestion other than `runtime_compilation_config.enabled` (`false` for backwards compatibility?)

Reply:
I wanted to stay away from that because if you set `runtime_compilation.enabled` to `false`, IMO the expectation would be that no runtime-compiled asset is going to be loaded - whereas what it _really_ means is that runtime compilation is still enabled, and assets which have been runtime compiled are going to be loaded, but the runtime compilation module which performs the compilation itself is going to be skipped

Reply:
Then something like `runtime_compilation_config.skip_compilation` would be better imo.